### PR TITLE
native: fix contract hashes

### PR DESCRIPTION
**Pull request by fyrchik**
They use fictive sender now.

### CODE REVIEW

File: pkg/core/blockchain.go [hunk 1]
```diff
@@ -1657,12 +1657,12 @@ var (
 func (bc *Blockchain) initVerificationVM(ic *interop.Context, hash util.Uint160, witness *transaction.Witness) error {
 	v := ic.VM
 	if len(witness.VerificationScript) != 0 {
-		if witness.ScriptHash() != hash {
-			return ErrWitnessHashMismatch
-		}
```
Comment (roman-khimov) [inline pkg/core/blockchain.go:1662]: Is this change really needed?

File: pkg/core/interop/context.go [hunk 2]
```diff
@@ -113,7 +114,7 @@ func NewContractMD(name string) *ContractMD {
 	emit.Syscall(w.BinWriter, interopnames.NeoNativeCall)
 
 	c.Script = w.Bytes()
-	c.Hash = hash.Hash160(c.Script)
+	c.Hash = state.CreateContractHash(hash.Hash160([]byte{byte(opcode.PUSH1)}), c.Script)
```
Comment (roman-khimov) [inline pkg/core/interop/context.go:117]: We better merge #1610 now first and then this one. And add some test that specifically checks for native contract's hashes and compares them to known values from C# devpack (see neo-project/neo-devpack-dotnet#391).

Comment (roman-khimov) [reply pkg/core/interop/context.go:117]: We can also add some method like `CreateNativeContractHash(name string)` to reduce code duplication between native contracts.

**Review by roman-khimov — CHANGES_REQUESTED** [review]

File: pkg/core/blockchain.go (hunk 1 above)
Comment (fyrchik) [reply pkg/core/blockchain.go:1662]: No, this was only to get a specific error in test. I think we can remove the second check now, while leaving the test.

File: pkg/core/interop/context.go [hunk 3]
```diff
@@ -119,7 +118,7 @@ func NewContractMD(name string) *ContractMD {
 	emit.Syscall(w.BinWriter, interopnames.SystemContractCallNative)
 
 	c.Script = w.Bytes()
-	c.Hash = hash.Hash160(c.Script)
+	c.Hash = state.CreateNativeContractHash(c.Script)
```
Comment (roman-khimov) [inline pkg/core/interop/context.go:121]: `CreateNativeContractHash(name string)`, move all these binwriter things into the function.

File: pkg/core/blockchain.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/core/blockchain.go:1645]: I don't see any test being changed, so we either need that change or don't need this one.

**Review by roman-khimov — CHANGES_REQUESTED** [review]
This is a breaking change and we need it to be merged ASAP.

File: pkg/core/interop/context.go (hunk 2 above)
Comment (fyrchik) [reply pkg/core/interop/context.go]: Took it directly from C# node, devpack has outdated values. Fixed.

File: pkg/core/interop/context.go (hunk 3 above)
Comment (fyrchik) [reply pkg/core/interop/context.go:121]: Fixed.

**Review by roman-khimov — APPROVED** [review]
